### ISSUE TITLE
Cannot add createdAt in @key fields  without making it available as part  of mutation

### ISSUE THREAD

**Issue by darshannahar** [aws-amplify/amplify-cli#4362]
**Describe the bug**
With @model directive createdAt is an autogenerated column in DynamoDB. I want to use it as a sort key. But I do not want to make it available in the mutation input. But amplify forces me to add createdAt as a part of the type and thus makes it available in mutation input.

**Amplify CLI Version**
4.20.0

**To Reproduce**
Graphql schema of below type won't be accepted:
```gql
type Comment @model @key(fields: ["blogId", "createdAt"]) {
  id: ID!
  content: String!
  blogId: ID!
  blog: Blog! @connection(fields: ["blogId"])
  customerId: ID!
}
```
**Expected behavior**
Should allow auto-added columns like createdAt, updatedAt to be part  of key

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version- 14.2.0

**Comment by dtelaroli** [comment]
+1

Same for ID

**Comment by darshannahar** [comment]
Duplicate of # 1657

**Comment by darshannahar** [comment]
Duplicate of #1657

**Comment by ammarkarachi** [comment]
@darshannahar we have a PR linked above closing it as it is a duplicate

**Comment by yuth** [comment]
This fix is released in Amplify CLI v4.21.1
